### ISSUE TITLE
Conflicts in names of actions from different modules

### ISSUE THREAD

**Issue by splinter89** [Codeception/Codeception#1181]
Hello!

There is a problem with current implementation of modules, and it's handling conflicting names of actions from different modules. Docs say that `The action from the second module will be loaded and the action from the first one will be ignored.` I see no flexibility for users here. This restriction prevents usage of conflicting modules in the same suite and sometimes leads to errors.

My proposal is to introduce some new logic of calling actions from modules. Let's say we have config like this:

``` yaml
modules:
    enabled: [Laravel4, Yii2]
```

Both modules have an action called `see`, and there is unique action called `seeInSession()` in `Laravel4` module. _Athough both modules implement `Framework`, so they have actions with common logic, they may have different configs, so results would be different as well._

Here is the concept:

``` php
// use fully qualified actions (it should always work as expected)
$I->viaLaravel4->see($text);
$I->viaYii2->see($text);

// both work the same
$I->viaLaravel->seeInSession($key);
$I->seeInSession($key);

// exception is thrown; user is asked to specify
// which module should be used to perform an action
$I->see($text);

// set the default module to use when resolving collisions
$I->useLaravel4();
$I->see($text1); // Laravel4::see() is called
$I->useYii2();
$I->see($text2); // Yii2::see() is called
```

As there may be multiple collisions, there should also be something like `$I->alsoUse...()`, which may be called multiple times to build a list of modules. And `$I->use...()` clears this list. For example, if `A` and `B` both have `ab()`, while `B` and `C` both have `bc()` action, ambiguity can be resolved like this:

``` php
$I->useA(); // A have the highest priority
$I->alsoUseB(); // if A is not involved in collision, B should take precedence
// following collisions will be resolved by [A, B]
$I->ab(); // A::ab()
$I->bc(); // B::bc()

$I->useA(); // [A]
$I->alsoUseC(); // [A, C]
$I->ab(); // A::ab()
$I->bc(); // C::bc()

$I->useC();
$I->alsoUseB(); // [C, B]
$I->ab(); // B::ab()
$I->bc(); // C::bc()

$I->useB(); // [B]
$I->alsoUseA(); // [B, A], but no effect
$I->alsoUseC(); // [B, A, C], but no effect
$I->ab(); // B::ab()
$I->bc(); // B::bc()
```

Of course, order of enabled modules in config will not play any role.

Helpers should also have similar support:

``` php
$I->viaSomeHelper->doAction(); // SomeHelper::doAction()
// or
$I->useSomeHelper();
$I->doAction();
$I->see($text);
```

As long as actions with the same name have the same signatures, there will be no problems with autocompletion: `build` command adds all methods (both duplicated _(just once)_ and unique) from all Modules and Helpers to Actor class, but also there are special properties for each module and Helper, like `viaLaravel4`, which hold an object of special class (just to help autocompletion) with methods that correspond to actions from `Laravel4` module exclusively.

Maybe it would be useful to use [traits](http://php.net/manual/en/language.oop5.traits.php) to represent modules.

_I used examples in Cept format for simplicity. Cest/Test formats should be supported as well._

To sum up:
- `$I->via{$Module}->action()`
- and `$I->via{$Module}->uniqueAction()`
- but also `$I->uniqueAction()`
- there is a priority list of modules, which should be used to resolve conflicts for `$I->action()`
- `$I->use{$Module}();` sets this list to `[$Module]`
- `$I->alsoUse{$Module}();` appends `$Module` to this list

I hope we'll find the best way out of this situation, so feel free to join the discussion and propose your solutions!
